src/librustc_mir/borrow_check/type_check/mod.rs

```diff
@@ -366,20 +366,6 @@ impl<'a, 'b, 'tcx> Visitor<'tcx> for TypeVerifier<'a, 'b, 'tcx> {
                         );
                     }
                 }
-            } else if let Some(static_def_id) = constant.check_static_ptr(tcx) {
```

Comment:
What is this diff doing here? This is changing the *borrow checker*, which should be unnecessary when all you want to do is change the dynamic const-to-static check.

Reply:
I guess I should have been clearer in the commit message. This check assumes that the only constants that can point to statics are the ones created and immediately dereferenced [during MIR building](https://github.com/rust-lang/rust/pull/71332/files#diff-33647ed7aa1d34b0fdc2c36d843a8169), so it fails with anything else. (For example, a constant pointing to a static's field, which has a different type than the static itself.)

So the first commit of this PR replaces this check with a more precise one suggested (AFAIU) by @eddyb [when it was added](https://github.com/rust-lang/rust/pull/69145/files#r382089824). I guess this could also make sense as a separate PR? Or at least some borrowck-oriented review?

Reply:
So without this, the rest of the PR fails, or so?

I certainly cannot review anything in borrw_check or hair, I have never touched or read any of that code.^^

Reply:
Yes, without this many scenarios using consts that refer to statics will fail at that `span_mirbug!`.